### ISSUE TITLE
Can't mock dynamically imported packages

### ISSUE THREAD

**Issue by christian-bromann** [vitest-dev/vitest#1294]
### Describe the bug

Scoped packages, e.g. `@wdio/logger` can't be mocked at all. In Jest it was possible to have the following structure:

```ts
__mocks__/
  - @wdio/
    - logger.ts
```

Jest would automatically pick this up.

### Reproduction

See https://stackblitz.com/edit/vitest-dev-vitest-tx14cb?file=package.json,test%2Fautomocking.spec.ts,__mocks__%2F%40wdio%2Flogger.ts&initialPath=__vitest__

I mocked out `@wdio/logger` in a way I expect Jest to work. The mocked file looks like this:
```ts
export default 'foobar';
```
In my test file (automocking.spec.ts) I do:
```ts
vi.mock('@wdio/test');
test('can automock scoped NPM package', () => {
  expect(foo).toBe('foobar');
});
```

The test failed due to:
```
can automock scoped NPM package
AssertionError: expected [Function getLogger] to be 'foobar' // Object.is equality
 - /test/automocking.spec.ts:38:41
``` 

### System Info

```shell
n/a
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
You misstyped:
```
vi.mock('@wdio/test');
// -> vi.mock('@wdio/logger');
```

**Comment by christian-bromann** [comment]
Uff my bad, sorry!

**Comment by christian-bromann** [comment]
@sheremet-va it turns out I made mistake reproducing the issue. The actual issue is that vitest fails in mocking dynamic imports. I updated the [stackblitz example](https://stackblitz.com/edit/vitest-dev-vitest-tx14cb?file=package.json,test%2Fautomocking.spec.ts,src%2Futils.ts&initialPath=__vitest).

If you have a method e.g.
```ts
export async function importTest(name: string) {
  const pkg = await import(name);
  return pkg;
}
```
and you want to test it, e.g.:

```ts
vi.mock('@wdio/logger');

test('can automock scoped NPM package', async () => {
  expect(await importTest('@wdio/logger')).toBe('foobar');
});
```

It fails with above mentioned error message. Is dynamic import mocks supported with vitest?
